File: contest/b_places_choisies/solutions/b_places_choisies.java
import java.util.Scanner;

public class b_places_choisies {

    public static void main(String[] args) {

        int n, h, c;
        Scanner sc = new Scanner(System.in);
        String temp;
        int currH = 0, currC = 0;

        n = sc.nextInt();
        h = sc.nextInt();
        c = sc.nextInt();

        for (int i = 0; (i < n) && (currC <= c) && (currH <= h); ++i){
            temp = sc.next();
            if(temp.equals("H")){
                ++currH;
            }else if (temp.equals("C")){
                ++currC;
            }
        }

        if ((currC > c) || (currH > h)){
            System.out.println("NON");
        }else{
            System.out.println("OUI");
        }
    }
}

File: contest/b_places_choisies/solutions/b_places_choisies.cs
using System;

namespace c_sharp
{
    internal class Program
    {
        public static void Main(string[] args)
        {
            var currH = 0; 
            var currC = 0;
            var n = int.Parse(Console.ReadLine());
            string[] line  = Console.ReadLine().Split(' ');
            var h = int.Parse(line[0]);
            var c = int.Parse(line[1]);
            string temp;
            
            for (int i = 0; (i < n) && (currC <= c) && (currH <= h); ++i){
                temp = Console.ReadLine();
                if(temp.CompareTo("H") == 0){
                    ++currH;
                }else if (temp.CompareTo("C") == 0){
                    ++currC;
                }
            }

            if ((currC > c) || (currH > h)){
                Console.WriteLine("NON");
            }else{
                Console.WriteLine("OUI");
            }
            
        }
    }
}
